src/publisher/ajson_ingestor.py

```diff
@@ -78,7 +78,7 @@ def _ingest(data, force=False):
         # only update the fragment if this article version has *not* been published *or* if force=True
         update_fragment = not av.published() or force
         fragments.add(av, XML2JSON, data['article'], pos=0, update=update_fragment)
-        fragments.set_article_json(av, quiet=True)
+        fragments.set_article_json(av, quiet=force) # article-json validation occurs here
```

Comment:
> Or is it because force should save the result no matter what?
correct. if you force something, you live with the results.

silent corrections are us saying: "modify a published article and don't tell anyone!", which feels pretty wrong. `force` neatly bypasses the rules, but there *is* that window for invalid ajson to get in.

to do things correctly here would require overhauling eLife's entire production process (no such thing as a 'silent correction', ever), and removing the `force` semantics seems very impractical (forcing an action may still fail), but a compromise could be:

`quiet=(settings.VALIDATE_ON_FORCE or force)`

so, if `VALIDATE_ON_FORCE` is `False`, and `force` is `True`, we preserve the very useful semantics of forcing an action.

if `VALIDATE_ON_FORCE` is `True`, a forced action may still fail BUT we have the certainty invalid articles can't sneak in this way, leaving us with just a changing spec causing invalidity.

I'm just trying to think if this flexibility is even useful to us or others ...